### PR TITLE
Add type hints to conditionals.

### PR DESCRIPTION
**Pull request by jesnie**
Add type hints to conditionals.

### CODE REVIEW

File: tests/gpflow/conditionals/test_multioutput.py [hunk 1]
```diff
@@ -23,38 +27,36 @@
 float_type = default_float()
 rng = np.random.RandomState(99201)
 
-
 # ------------------------------------------
 # Helpers
 # ------------------------------------------
 
 
-def predict(model, Xnew, full_cov, full_output_cov):
```
Comment (uri-granta) [inline tests/gpflow/conditionals/test_multioutput.py:32]: (what an odd function!)

File: tests/gpflow/conditionals/test_multioutput.py [hunk 2]
```diff
@@ -23,38 +27,36 @@
...
     for model in models:
-        m, v = predict(model, Xnew, full_cov, full_output_cov)
+        m, v = model.predict_f(Xnew, full_cov=full_cov, full_output_cov=full_output_cov)
         ms.append(m)
         vs.append(v)
     return ms, vs
 
 
-def assert_all_array_elements_almost_equal(arr, decimal):
+def assert_all_array_elements_almost_equal(arr: Sequence[tf.Tensor]) -> None:
```
Comment (uri-granta) [inline tests/gpflow/conditionals/test_multioutput.py:49]: Do you think the original intent might have been to set `atol` to `0.1**decimal`?

**Review by uri-granta — APPROVED** [review]
Looks good!

File: tests/gpflow/conditionals/test_multioutput.py (hunk 2 above)
Comment (jesnie) [reply tests/gpflow/conditionals/test_multioutput.py:49]: > Do you think the original intent might have been to set `atol` to `0.1**decimal`?

I think the original intent was to do `np.testing.assert_almost_equal(..., decimal=decimal)`.
Apparently that turned into `np.testing.asert_allclose(..., atol=1e-5)`.
:shrug:

**Pull request merged**